### ISSUE TITLE
pg_dump fails on master after 9.2 merge for complex schemas

### ISSUE THREAD

**Issue by chrishajas** [greenplum-db/gpdb#5697]
After the 9.2 merge, a schema-only pg_dump fails on a complex schema with the following error: 

Output:
```
pg_dump: [archiver (db)] query failed: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.

Unexpected exception reached top of execution stack: major=200 minor=14 file=CTranslatorUtils.cpp line=164",,,,,,,,"SysLoggerMain","syslogger.c",682,
2018-09-05 22:11:18.782640 UTC,"gpadmin","metadatascaledb",p27022,th-1363257568,"[local]",,2018-09-05 22:11:18 UTC,0,con6,cmd22,seg-1,,dx10,,sx1,"LOG","00000","Planner produced plan :0",,,,,,"SELECT c.tableoid, c.oid, c.relname, c.relacl, c.relkind, c.relstorage, c.relnamespace, (SELECT rolname FROM pg_catalog.pg_roles WHERE oid = c.relowner) AS rolname, c.relchecks, c.relhastriggers, c.relhasindex, c.relhasrules, c.relhasoids, c.relfrozenxid, tc.oid AS toid, tc.relfrozenxid AS tfrozenxid, c.relpersistence, CASE WHEN c.reloftype <> 0 THEN c.reloftype::pg_catalog.regtype ELSE NULL END AS reloftype, d.refobjid AS owning_tab, d.refobjsubid AS owning_col, (SELECT spcname FROM pg_tablespace t WHERE t.oid = c.reltablespace) AS reltablespace, array_to_string(c.reloptions, ', ') AS reloptions, array_to_string(array(SELECT 'toast.' || x FROM unnest(tc.reloptions) x), ', ') AS toast_reloptions , p.parrelid as parrelid,  pl.parlevel as parlevel FROM pg_class c LEFT JOIN pg_depend d ON (c.relkind = 'S' AND d.classid = c.tableoid AND d.objid = c.oid AND d.objsubid = 0 AND d.refclassid = c.tableoid AND d.deptype = 'a') LEFT JOIN pg_class tc ON (c.reltoastrelid = tc.oid) LEFT JOIN pg_partition_rule pr ON c.oid = pr.parchildrelid LEFT JOIN pg_partition p ON pr.paroid = p.oid LEFT JOIN pg_partition pl ON (c.oid = pl.parrelid AND pl.parlevel = 0)WHERE c.relkind in ('r', 'S', 'v', 'c', 'f') AND c.oid NOT IN (SELECT p.parchildrelid FROM pg_partition_rule p LEFT JOIN pg_exttable e ON p.parchildrelid=e.reloid WHERE e.reloid IS NULL)ORDER BY c.oid",0,,"orca.c",61,
2018-09-05 22:11:35.538244 UTC,,,p27002,th-1363257568,,,,0,,,seg-1,,,,,"LOG","00000","FTS: segment (content=0, dbid=2, role=p) reported isMirrorUp 1, isInSync 1, isSyncRepEnabled 1, isRoleMirror 0, and retryRequested 0 to the prober.",,,,,,,0,,"ftsprobe.c",625,
2018-09-05 22:11:35.538322 UTC,,,p27002,th-1363257568,,,,0,,,seg-1,,,,,"LOG","00000","FTS: segment (content=1, dbid=3, role=p) reported isMirrorUp 1, isInSync 1, isSyncRepEnabled 1, isRoleMirror 0, and retryRequested 0 to the prober.",,,,,,,0,,"ftsprobe.c",625,
2018-09-05 22:11:46.427856 UTC,,,p27022,th0,,,2018-09-05 22:11:18 UTC,0,con6,cmd23,seg-1,,,,,"PANIC","XX000","Unexpected internal error: Master process received signal SIGSEGV",,,,,,,0,,,,"1    0xa35e75 postgres StandardHandlerForSigillSigsegvSigbus_OnMainThread (elog.c:4923)
2    0x384020f7e0 libpthread.so.0 <symbol not found> + 0x4020f7e0
3    0x717299 postgres ExecHashJoin (nodeHashjoin.c:333)
4    0x6f2b93 postgres ExecProcNode (execProcnode.c:1051)
5    0x721465 postgres ExecNestLoop (tuptable.h:155)
6    0x6f2bb3 postgres ExecProcNode (execProcnode.c:1043)
7    0x717659 postgres ExecHashJoin (tuptable.h:155)
8    0x6f2b93 postgres ExecProcNode (execProcnode.c:1051)
9    0x724900 postgres ExecSort + 0x1f0
10   0x6f2b73 postgres ExecProcNode (execProcnode.c:1062)
11   0x6e9c49 postgres <symbol not found> (tuptable.h:155)
12   0x6ea2dc postgres standard_ExecutorRun (execMain.c:988)
13   0x8ce177 postgres <symbol not found> (pquery.c:0)
14   0x8d04c1 postgres PortalRun + 0x341
15   0x8ca71e postgres <symbol not found> (postgres.c:1810)
16   0x8ccccb postgres PostgresMain (postgres.c:5194)
"
```

Repro:
```
SET enable_nestloop TO on;
SELECT c.tableoid, c.oid, c.relname, c.relacl, c.relkind, c.relstorage, c.relnamespace, (SELECT rolname FROM pg_catalog.pg_roles WHERE oid = c.relowner) AS rolname, c.relchecks, c.relhastriggers, c.relhasindex, c.relhasrules, c.relhasoids, c.relfrozenxid, tc.oid AS toid, tc.relfrozenxid AS tfrozenxid, c.relpersistence, CASE WHEN c.reloftype <> 0 THEN c.reloftype::pg_catalog.regtype ELSE NULL END AS reloftype, d.refobjid AS owning_tab, d.refobjsubid AS owning_col, (SELECT spcname FROM pg_tablespace t WHERE t.oid = c.reltablespace) AS reltablespace, array_to_string(c.reloptions, ', ') AS reloptions, array_to_string(array(SELECT 'toast.' || x FROM unnest(tc.reloptions) x), ', ') AS toast_reloptions , p.parrelid as parrelid,  pl.parlevel as parlevel FROM pg_class c LEFT JOIN pg_depend d ON (c.relkind = 'S' AND d.classid = c.tableoid AND d.objid = c.oid AND d.objsubid = 0 AND d.refclassid = c.tableoid AND d.deptype = 'a') LEFT JOIN pg_class tc ON (c.reltoastrelid = tc.oid) LEFT JOIN pg_partition_rule pr ON c.oid = pr.parchildrelid LEFT JOIN pg_partition p ON pr.paroid = p.oid LEFT JOIN pg_partition pl ON (c.oid = pl.parrelid AND pl.parlevel = 0)WHERE c.relkind in ('r', 'S', 'v', 'c', 'f') AND c.oid NOT IN (SELECT p.parchildrelid FROM pg_partition_rule p LEFT JOIN pg_exttable e ON p.parchildrelid=e.reloid WHERE e.reloid IS NULL)ORDER BY c.oid;
```

If I set `enable_nestloop` to off, this query completes. There is another instance in pg_dump where we set `enable_nestloop` to off temporarily for performance reasons(c4a7b25143d727d01a7545e6c4a29c5e6edbd90c). Is setting `enable_nestloop` to on during pg_dump still beneficial to pg_dump's performance?

**Comment by hlinnaka** [comment]
Can you narrow this down to a fully reproducable test case? I tried that on the 'regression' database, after running the regression test suite, but could not reproduce.

**Comment by chrishajas** [comment]
I have a ~170MB schema that it repros reliably on, which I'd be happy to share privately, but it's quite large and takes a couple hours to load. It has some massive partition tables (~500k lines of DDL for a single partition table), which is probably the culprit.

**Comment by Eulerizeit** [comment]
@chrishajas can that DDL be shared for us to Debug this?

**Comment by chrishajas** [comment]
This seems to have resolved itself after the 9.3 merge.
